controllers/machineset_controller.go

```diff
@@ -445,7 +447,7 @@ func (r *MachineSetReconciler) MachineToMachineSets(o handler.MapObject) []ctrl.
 
 	m, ok := o.Object.(*clusterv1.Machine)
 	if !ok {
-		klog.Errorf("expected a Machine but got a %T", o.Object)
+		r.Log.Error(errors.New("did not get machine"), "got", fmt.Sprintf("%T", o.Object))
```

Comment:
We can probably leave the error as `nil` and just provide the message `expected to receive a machine object type` and the key value can be `ObjectType`.
See [logging errors](https://github.com/kubernetes-sigs/controller-runtime/blob/aac8bc7a3cbc56b99cc250d1506a0a39765b573d/TMP-LOGGING.md#logging-errors).

Reply:
thanks! I've updated it with nil error, ptal :)